### ISSUE TITLE
Overload of interpreters has disturbing effect on documentation pages

### ISSUE THREAD

**Issue by Yewzir** [RockstarLang/rockstar#364]
Greetings @dylanbeattie

First and foremost, congratulations on releasing the newest Rockstar implementation running on Microsoft .NET, quite an improvement.

Now, straight to business. I can't speak for anyone else really, but I think there's an overload of interpreters to be found on the documentation pages. Practically every section per topic goes with an interpreter at the end. I wonder, is that really necessary? To me personally, the disturbing effect is enormous, and I find it really hard to focus on things I'm looking for because there's always an interpreter that catches the eye. There's at least one page with seventeen of these things, why? For instance, most of the examples already contain results of what's about to be printed in their comments. I mean, we believe you, really. In my honest opinion, I don't think the documentation pages should contain any of these interpreters. Another idea is to provide just one in a fixed position, allowing users to toggle its visibility. I'm positive it'll make the overall browsing experience better, way better.

Kind regards
@Yewzir

**Comment by caseyc37** [comment]
Huh.

Personally, I like the interpreters on the documentation pages. It means that if I want to try some slight variation on the described behaviour, there's a nice testbed right there to mess around with. Very convenient.

**Comment by Yewzir** [comment]
@caseyc37

Two posts, two times "personally". Good stuff, really. Never have I said everyone'd agree, which isn't what this is even about. I'd just like to hear the repository owner's reasoning, hence the tag. Regardless, if you want to speak, you will speak. If I want to speak, I will speak.

Peace, and rock the fuck on!

Regards
@Yewzir

**Comment by dylanbeattie** [comment]
@Yewzir I asked some of the other folks in the Rockstar what they thought about your suggestion. We've had a few dozen people using Rockstar extensively for Advent of Code puzzles since the new release shipped, and you're the first person to comment on the embedded interpreters in the docs; I wanted to get some perspective from other folks using the new version on whether they find them useful or not.

**Comment by Yewzir** [comment]
> @Yewzir I asked some of the other folks in the Rockstar what they thought about your suggestion. We've had a few dozen people using Rockstar extensively for Advent of Code puzzles since the new release shipped, and you're the first person to comment on the embedded interpreters in the docs; I wanted to get some perspective from other folks using the new version on whether they find them useful or not.

This isn't about them being _useless_. Not by a long shot. I'm only trying to understand why there are so many. Did you make a change by any chance? They are hidden by default now, with a _Try It_ toggle. Like that, you can keep all of them, and it's perfect! Cheers, Dylan!

**Comment by mikachelya** [comment]
Don't think it's a big issue.
my only complaint with those is that they seem to break the browser find in page function. On firefox, If you search for a word that is both in interpreters and in the page itself, you can't really click through the matches. It's hard to describe, i can post a video on the discord if you need

**Comment by dylanbeattie** [comment]
I just pushed a change that hides the interpreters behind a toggle.

Every code sample in the docs is a Rockstar file that's included using a custom Jekyll plugin when the site is built, and the Rockstar build process actually runs all the documentation samples as part of the site build so if there are bugs in the docs it'll fail the build; I added the play/parse/reset buttons to the embedded code editor early on in the process to make the docs interactive, and then added the command-line args and STDIN inputs.

I agree they look much cleaner hidden behind the toggle. Thanks for the suggestion.
